webapp/IronCalc/src/components/WorkbookSettings/WorkbookSettingsDialog.tsx

```diff
@@ -0,0 +1,429 @@
+import styled from "@emotion/styled";
+import {
+  Autocomplete,
+  Box,
+  Dialog,
+  FormControl,
+  MenuItem,
+  Select,
+  TextField,
+} from "@mui/material";
+import { Check, X } from "lucide-react";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { theme } from "../../theme";
+
+type WorkbookSettingsDialogProps = {
+  open: boolean;
+  onClose: () => void;
+  initialLocale: string;
+  initialTimezone: string;
+  onSave?: (locale: string, timezone: string) => void;
+};
+
+const WorkbookSettingsDialog = (properties: WorkbookSettingsDialogProps) => {
+  const { t } = useTranslation();
+  const locales = ["en-US", "en-GB", "de-DE", "fr-FR", "es-ES"];
+  const timezones = [
+    "Berlin, Germany (GMT+1)",
+    "New York, USA (GMT-5)",
+    "Tokyo, Japan (GMT+9)",
+    "London, UK (GMT+0)",
+    "Sydney, Australia (GMT+10)",
+  ];
```

Comment:
There is no reason for us not to show all the timezones in the world. I _think_ I would like to take them from the model.

In modern js you can take all of them by:

```
const timezones = Intl.supportedValuesOf("timeZone");
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/supportedValuesOf